projects/vdk-plugins/vdk-notebook/tests/test_plugin.py

```diff
@@ -1,30 +1,64 @@
+from vdk.plugin.sqlite import sqlite_plugin
 from vdk.plugin.test_utils.util_funcs import cli_assert_equal
 from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
 from vdk.plugin.test_utils.util_funcs import jobs_path_from_caller_directory
 
-"""
-This is a sample test file showing a recommended way to test new plugins.
-A good way to test a new plugin is how it would be used in the command that it extends.
-"""
 
+@mock.patch.dict(
+    os.environ,
+    {
+        "VDK_DB_DEFAULT_TYPE": "SQLITE",
+        "VDK_INGEST_METHOD_DEFAULT": "sqlite",
+    },
+)
+class JupyterTests(unittest.TestCase):
+    def setUp(self) -> None:
+        self.__runner = CliEntryBasedTestRunner(notebook_plugin, sqlite_plugin)
 
-def test_dummy():
-    with mock.patch.dict(
-        os.environ,
-        {
-            # mock the vdk configuration needed for our test
-        },
-    ):
-        # CliEntryBasedTestRunner (provided by vdk-test-utils) gives a away to simulate vdk command
-        # and mock large parts of it - e.g passed our own plugins
-        runner = CliEntryBasedTestRunner(notebook_plugin)
+    def test_notebook_plugin(self) -> None:
+        result: Result = self.__runner.invoke(
+            ["run", jobs_path_from_caller_directory("rest-api-job")]
+        )
+        cli_assert_equal(0, result)
+        actual_rs: Result = self.__runner.invoke(
+            ["sqlite-query", "--query", f"SELECT * FROM rest_target_table"]
+        )
+        assert actual_rs.stdout == (
+            "  userId    id  title                 completed\n"
+            "--------  ----  ------------------  -----------\n"
+            "       1     1  delectus aut autem            0\n"
+        )
 
-        # result: Result = runner.invoke(["run", jobs_path_from_caller_directory("job-using-a-plugin")])
-        # cli_assert_equal(0, result)
+    def test_failing_job_with_syntax_error(self) -> None:
+        result: Result = self.__runner.invoke(
+            ["run", jobs_path_from_caller_directory("rest-api-job-fail-syntax-error")]
+        )
+        cli_assert_equal(1, result)
+
+    def test_failing_job_with_code_error(self) -> None:
+        result: Result = self.__runner.invoke(
+            ["run", jobs_path_from_caller_directory("rest-api-job-fail-code-error")]
+        )
+        cli_assert_equal(2, result)
+
+    def test_notebook_plugin_with_multiple_run_methods(self) -> None:
+        result: Result = self.__runner.invoke(
+            ["run", jobs_path_from_caller_directory("rest-api-multiple-run-methods")]
+        )
+        cli_assert_equal(0, result)
+        actual_rs: Result = self.__runner.invoke(
+            ["sqlite-query", "--query", f"SELECT * FROM rest_target_table_copy"]
+        )
+        assert actual_rs.stdout == (
+            "  userId    id  title                 completed\n"
+            "--------  ----  ------------------  -----------\n"
+            "       1     1  delectus aut autem            0\n"
+        )
```

Comment:
what is the expectation about the mixed file types use-case scenario `(.ipynb +[.sql | .py])`?

Reply:
currently, we won't have mixed data jobs.

Reply:
Okay, however is it possible for a user to mix those files, and what is expected to happen?